renku/core/commands/dataset.py

```diff
@@ -626,6 +657,23 @@ def import_dataset(
             total_size=total_size,
         )
 
+        if previous_dataset:
+            previous_dataset.update_metadata(dataset)
+            current_files = set(f.url for f in new_files)
+            # remove files not present in the dataset anymore
+            for f in previous_dataset.files:
+                if f.url in current_files:
+                    continue
+
+                previous_dataset.unlink_file(f.path)
+
+                if delete:
+                    client.remove_file(client.path / f.path)
+
+            dataset = previous_dataset
+            client.mutate_dataset(dataset)
+            dataset.to_yaml()
```

Comment:
Actually this check won't work for Renku imports:
```
             current_files = set(f.url for f in new_files)
             # remove files not present in the dataset anymore
             for f in previous_dataset.files:
                 if f.url in current_files:
                     continue
```
because url for all files is the dataset's url. We need to check the path there.